### DISCUSSION TITLE
groovy oneliner to get first token/column of tsv file

### DISCUSSION THREAD

**Discussion by colindaven** [nextflow-io/nextflow#2884] [Q&A] [answered]
Dear all,

this is dsl 2.0

I know I can use file(somefile.fai).readLines() to read a whole file. If the file only contains one column (chromosome name) this works great.

    fai_channel = Channel
        //.fromList(file(params.fai).readLines())
        //.fromList(file(prepare_fai.out).readLines())
        //.fromList((prepare_fai.out).readLines())
        .fromList(file(params.fai).readLines().splitEachLine("\t"))
        .map { [ it ] }
        .view()
   

But can I just read the first column of the tab separated TSV in a one liner? Eg split by tab, then get the 0th token.

I've tried using a process to prepare a multi-column TSV (fai file) into a single column  one using
`       cut -f 1 x.fai > x.fai.txt`
, but then I can't seems to read it using file(prepare_fai.out) above.

If not, would another method (eg value channel) be compatible with the .map I want to do here ?

Thanks!
Colin

**Comment by phue** [comment, marked as answer]
Hi Colin,

`.splitText()` allows to specify a closure, so something like this should to the trick:

```
Channel
 .fromPath(params.fai)
 .splitText( each:{ it.split().first() } )
 .view()
```

Cheers,
Patrick

**Reply by colindaven** [reply]
Thanks Patrick, that worked a treat. Now I guess I'll have to read up a bit more on groovy closures ... 
cheers,
Colin
